docs/root/configuration/best_practices/edge.rst

```diff
@@ -1,5 +1,91 @@
+(to avoid consuming HTTP headers from external clients, see :ref:`HTTP header sanitizing <config_http_conn_man_header_sanitizing>` for details),
+and HTTP/2 settings (e.g.
+:ref:`HTTP/2 maximum concurrent streams limit <envoy_api_field_core.Http2ProtocolOptions.max_concurrent_streams>` to 100,
+:ref:`HTTP/2 initial stream window size limit <envoy_api_field_core.Http2ProtocolOptions.initial_stream_window_size>` to 64 KiB,
+and :ref:`HTTP/2 initial connection window size limit <envoy_api_field_core.Http2ProtocolOptions.initial_connection_window_size>` to 1 MiB).
+
+The following is a YAML example of the above recommendation.
+
+.. code-block:: yaml
+
+  overload_manager:
+    refresh_interval: 0.25s
+    resource_monitors:
+    - name: "envoy.resource_monitors.fixed_heap"
+      config:
+        # TODO: Tune for your system.
+        max_heap_size_bytes: 2147483648 # 2 GiB
+    actions:
+    - name: "envoy.overload_actions.shrink_heap"
+      triggers:
+      - name: "envoy.resource_monitors.fixed_heap"
+        threshold:
+          value: 0.95
+    - name: "envoy.overload_actions.stop_accepting_requests"
+      triggers:
+      - name: "envoy.resource_monitors.fixed_heap"
+        threshold:
+          value: 0.98
+
+  static_resources:
+    listeners:
+    - address:
+        socket_address: { address: 127.0.0.1, port_value: 443 }
+      listener_filters:
+      - name: "envoy.listener.tls_inspector"
+        typed_config: {}
+      per_connection_buffer_limit_bytes: 32768 # 32 KiB
+      filter_chains:
+      - filter_chain_match:
+          server_names: ["example.com", "www.example.com"]
+        tls_context:
+          common_tls_context:
+            tls_certificates:
+            - certificate_chain: { filename: "example_com_cert.pem" }
+              private_key: { filename: "example_com_key.pem" }
+        filters:
+        - name: envoy.http_connection_manager
+          typed_config:
+            "@type": type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
+            stat_prefix: ingress_http
+            use_remote_address: true
+            common_http_protocol_options:
+              idle_timeout: 840s
+            http2_protocol_options:
+              max_concurrent_streams: 100
+              initial_stream_window_size: 65536 # 64 KiB
+              initial_connection_window_size: 1048576 # 1 MiB
+            route_config:
+              virtual_hosts:
+              - name: default
+                domains: "*"
+                routes:
+                - match: { prefix: "/" }
+                  route: { cluster: service_foo }
+    clusters:
+      name: service_foo
+      connect_timeout: 15s
+      per_connection_buffer_limit_bytes: 32768 # 32 KiB
+      hosts:
+        socket_address:
+          address: 127.0.0.1
+          port_value: 8080
+      http2_protocol_options:
+        max_concurrent_streams: 100
```

Comment:
I'm not sure that we want this for upstream. IMO the default is fine and have circuit breakers to limit this on a per-cluster basis as well as the downstream limits.

Reply:
What is the scope here? Do we want to over edge use cases where Envoy is acting as an egress gateway and speaking to untrusted backends? In this situation, we also need to be defensive on the upstream.

Reply:
We don't support push, so this provides no protection against the upstream. IIRC it might also limit the number of outgoing streams, but that is already controlled by downstream limits and circuit breakers.

Reply:
Right. I guess I'd suggest that we do configure all the other defensive limits that we have, or at least provide some nuanced discussion on trust model here. In fact, I think there should be a short section on trust model. By edge, are we really saying "edge ingress"? Because, for "edge egress", we know today that there are some missing features that came up during the HTTP/2 headers CVE handling for example, that are needed when dealing with untrusted upstreams.

Reply:
Agreed, let's be explicit and also call out a warning against untrusted upstream.

Reply:
What do you want to see here? It's unclear to me from the comments.

Reply:
I would delete this line. I don't think it does anything useful.